### PR TITLE
Update roman for release CAL version 0.8.1

### PR DESCRIPTION
**Pull request by jaytmiller**
Stripped down roman build for v0.8.1 and GitHub dev releases.
Added distlib to common-env/testing.pip for get-imports tool.
Added update-requirements to roman-cal, and more generally to image-update when defined. 
Added CAL_VERSION env setting to setup-env.template, defaulting to latest. 
Added automatic kernel specific subtests,  and test.disabled for roman-cal based on github. 
Remove versions from common packages in favor of putting all version constraints in mission specific build-hints.pip. Residual Safety cleanup
Modified image-test + env-test system to support environment-specific `test` script.
Added test.disabled to roman-cal environment for running romancal GitHub unit tests,  requires CRDS and/or artifactory.
Modified env-conda to add .explicit handling for @explicit format conda files given by CAL releases.
Modified GitHub Actions to drop constant FROZEN, CHILLY builds on PR/PUSH to main.
Modified GitHub Actions to add CAL_VERSION variable, roman 0.8.1(none) and latest builds.
Start on automatic get-imports tool to map packages to test imports;  updated roman-cal imports test list. 
Froze roman requirements for CAL_VERSION=0.8.1;  offsite updates cannot access artifactory lists,  github incomplete,  laptop updates only. 
Updated FRAMEWORK.md

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -9,15 +9,21 @@ on:
    workflow_dispatch:
 jobs:
   docker:
-    name: "${{ matrix.DEPLOYMENT_NAME }}  USE_FROZEN=${{ matrix.USE_FROZEN }} -- Image Build, Test"
+    name: "${{ matrix.DEPLOYMENT_NAME }}  USE_FROZEN=${{ matrix.USE_FROZEN }} ${{ matrix.CAL_VERSION }} -- Image Build, Test"
```
Comment (cslocum) [inline .github/workflows/build-test.yml:12]: Probably a dumb question, but what is matrix here?

File: deployments/roman/image/environments/roman-cal/roman-cal.explicit [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+https://repo.anaconda.com/pkgs/main/linux-64/xz-5.2.5-h7f8727e_1.conda
+https://repo.anaconda.com/pkgs/main/linux-64/zlib-1.2.12-h7f8727e_2.conda
+https://repo.anaconda.com/pkgs/main/linux-64/readline-8.1.2-h7f8727e_1.conda
+https://repo.anaconda.com/pkgs/main/linux-64/tk-8.6.12-h1ccaba5_0.conda
+https://repo.anaconda.com/pkgs/main/linux-64/sqlite-3.39.2-h5082296_0.conda
+https://repo.anaconda.com/pkgs/main/linux-64/python-3.9.12-h12debd9_1.conda
+https://repo.anaconda.com/pkgs/main/linux-64/certifi-2022.6.15-py39h06a4308_0.conda
+https://repo.anaconda.com/pkgs/main/noarch/wheel-0.37.1-pyhd3eb1b0_0.conda
+https://repo.anaconda.com/pkgs/main/linux-64/setuptools-63.4.1-py39h06a4308_0.conda
+https://repo.anaconda.com/pkgs/main/linux-64/pip-22.1.2-py39h06a4308_0.conda
```
Comment (cslocum) [inline deployments/roman/image/environments/roman-cal/roman-cal.explicit:26]: What was the motivation for installing these with conda?

**Review by cslocum — APPROVED** [review]
This is awesome, really nice.  I left a few comments just out of curiosity, but not blockers.

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (jmatuskey) [reply .github/workflows/build-test.yml:12]: if you look at matrix a few lines under, it basically will run one job for each permutation of all the different matrices. so it'll run name: roman , use_frozen = 0 then name: roman, use_frozen = 1, then name: tike, use_frozen = 0 etc.

Comment (jmatuskey) [reply .github/workflows/build-test.yml:12]: am I correct in assuming that you removed chilly build from the matrix to cut down on it running so many extra jobs?

**Review by jmatuskey — APPROVED** [review]

File: .github/workflows/build-test.yml (hunk 1 above)
Comment (jaytmiller) [reply .github/workflows/build-test.yml:12]: Yeah,  and frozen.   We need to switch to a mode where we only freeze/chill in the post-merge build tests;  so developers can be hands-off and mostly forget about freezing personally,  and we don't pay for 9 builds on every PR update.   CodeBuild for roman will be really expensive unless we're careful.  But running automated tests on the hub should mitigate that some,  if we can work that out.

**Pull request merged**